lib/shared/src/prompt/truncation.ts

```diff
@@ -50,6 +52,9 @@ export function truncateTextNearestLine(
  * Truncates text to the given number of tokens, keeping the end of the text.
  */
 export function truncateTextStart(text: string, maxTokens: number): string {
-    const maxLength = maxTokens * CHARS_PER_TOKEN
-    return text.length <= maxLength ? text : text.slice(-maxLength - 1)
+    const encoded = TokenCounter.encode(text)
+    if (encoded.length < maxTokens) {
+        return text
+    }
+    return TokenCounter.decode(encoded.slice(-maxTokens - 1))
```

Comment:
This should be `-maxTokens`, you'll get exactly `maxTokens` (e.g. `'food'.slice(-2)` is `'od'`)
```suggestion
    return TokenCounter.decode(encoded.slice(-maxTokens))
```